src/Microsoft.Data.Analysis/DataFrame.Arrow.cs

```diff
@@ -137,14 +151,18 @@ public IEnumerable<RecordBatch> ToArrowRecordBatches()
             Apache.Arrow.Schema.Builder schemaBuilder = new Apache.Arrow.Schema.Builder();
 
             int columnCount = Columns.Count;
+            List<Field> fields = new List<Field>(columnCount);
             for (int i = 0; i < columnCount; i++)
             {
                 DataFrameColumn column = Columns[i];
                 Field field = column.GetArrowField();
-                schemaBuilder.Field(field);
+                fields.Add(field);
             }
 
-            Schema schema = schemaBuilder.Build();
+            StructType structType = new StructType(fields);
```

Comment:
Wrapping the arrays in a `StructType` feels too much like Spark-specific functionality here to me. I think we should move that logic to dotnet/spark instead of having it in here.

My thinking is that if I had a `DataFrame` with 5 columns, I should get a `RecordBatch` with 5 columns. Not 1 `StructType` column with 5 fields.

Reply:
Removed this now.